src/VisualStudio/Core/Test/CodeModel/CSharp/CodeEventTests.vb

```diff
@@ -221,6 +253,38 @@ class C
                      TextPoint(line:=7, lineOffset:=6, absoluteOffset:=96, lineLength:=5)))
         End Sub
 
+        <WorkItem(2437, "https://github.com/dotnet/roslyn/issues/2437")>
+        <ConditionalFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Sub GetEndPointExplicitlyImplementedEvent()
+            Dim code =
+<Code>
+delegate void SampleEventHandler(object sender);
+
+interface I1
+{
+    event SampleEventHandler SampleEvent;
+}
+
+class C1 : I1
+{
+    event SampleEventHandler $$I1.SampleEvent
+    {
+        add
+        {
+        }
+
+        remove
+        {
+        }
+    }
+}
+</Code>
+
+            TestGetEndPoint(code,
+                Part(EnvDTE.vsCMPart.vsCMPartWholeWithAttributes,
```

Comment:
What about some tests here for `vsCMPartName`, since that is the thing that is interesting about explicit interface implementations.